### ISSUE TITLE
Inconsistent formatting in heredocs between 'nix and Windows

### ISSUE THREAD

**Issue by paulanthonywilson** [elixir-lang/elixir#13159]
### Elixir and Erlang/OTP versions

```
Erlang/OTP 26 [erts-14.1] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit:ns]

Elixir 1.15.6 (compiled with Erlang/OTP 26)
```

and

```
Erlang/OTP 25 [erts-13.2.2.4] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit:ns]

Elixir 1.14.2 (compiled with Erlang/OTP 24)
```

### Operating system

Windows 10 and Linux (mixed preferences and build)

### Current behavior

Relates to https://github.com/elixir-lang/elixir/issues/10504 and https://github.com/elixir-lang/elixir/commit/94b3539527bb697e4b12b746c206a1ec0a623447. (Also relates to git, which is outwith the scope of an Elixir issue).

## Scene 1: a programmer is programming Elixir using Windows using default git config

When running `mix format` on a DOS style file then all carriage returns (`0x0d`) are stripped except those in heredocs. This is as intended.  `mix format --check-formatted` is happy enough, as you would expect having just formatted the files. The code is committed to a remote git repository.

(Vim style carriage returns denoted by ^M spaces by dots):

```elixir
defmodule Lineend do
··@moduledoc """
··Documentation for `Lineend`.^M
··^M
··This has two paragraphs.^M
  """

··def hello do
····:world
··end
end
```

## Scene 2: another programmer is on Linux (or maybe it's a CI build) using that default git config

The programmer (or build) checks out the files. Because this is Linux then the files  have been checkout with with carriage returns.

```elixir
defmodule Lineend do
··@moduledoc """
··Documentation for `Lineend`.
··
··This has two paragraphs
  """

··def hello do
····:world
··end
end
```

`mix format --check-formatted` fails because of the trailing whitespace between the paragraphs in the heredoc.

### Expected behavior

I am not super keen on this retention of carriage returns in the heredoc. It gets inconsistent between Operating Systems when we throw git into the mix. We could blame git(*) but mostly we have to live in that world. If we really wanted carriage returns in the heredoc we could use `\r`. But we are where we are.

Really I think the formatting should not treat the `0xd` as non-whitespace character and indent it when retained.

```elixir

defmodule Lineend do
··@moduledoc """
··Documentation for `Lineend`.^M
^M
··This has two paragraphs^M
```

The workaround is to get the Windows programmers to `git config --global core.autocrlf input`. That is harder than you might think, especially if they're using all kinds of weird and wonderful Git GUIs.

(*) `git blame`

**Comment by simonmcconnell** [comment]
adding a `.gitattributes` and optionally a `.editorconfig` to the repository avoids this and doesn't require individuals to have the correct config.  See the livebook repo:

https://github.com/livebook-dev/livebook/blob/f0979ea4922903397b523d14d0b04f1f4ab26e9d/.gitattributes#L1C1-L2C19
https://github.com/livebook-dev/livebook/blob/main/.editorconfig

https://git-scm.com/docs/gitattributes
https://editorconfig.org/

**Comment by paulanthonywilson** [comment]
> adding a `.gitattributes` and optionally a `.editorconfig`

That is excellent to know. Thank you.

**Comment by josevalim** [comment]
I am 99% sure this has been fixed on recent Elixir versions. Can you please try 1.6.0-rc?

**Comment by paulanthonywilson** [comment]
Ah, you're right. It is fixed on `1.16.0-rc.0`. Carriage returns are no longer stripped either.

Sorry for not trying on that release candidate. I had it in my head that I would have to try it out on Windows and couldn't as easily switch Elixir versions as I can on linux or OS X with asdf. I was wrong and could easily give it a shot on my Mac by injecting carriage returns (with .gitattributes).
